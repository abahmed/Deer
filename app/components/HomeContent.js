import React, { Component } from 'react'
import PropTypes from 'prop-types'
import NoteEditor from './../containers/NoteEditor'

export default class HomeContent extends Component {
  render () {
    // Show homeContent when no note is selected.
    if (this.props.activeNoteIndex >= 0) { return (<NoteEditor />) }

    return (
      <div className='center-text middle-page'>
        <h4>Please select a note or add a new one</h4>
      </div>
    )
  }
}

HomeContent.propTypes = {
  activeNoteIndex: PropTypes.number.isRequired
}
