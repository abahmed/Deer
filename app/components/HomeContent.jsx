import React, { Component } from 'react'
import PropTypes from 'prop-types'
import NoteEditor from '../containers/NoteEditor'

export default class HomeContent extends Component {
  render () {
    const { notes, t } = this.props
    const isAtLeastOneNote = notes.length > 0

    // Show homeContent when no note is selected.
    if (this.props.activeNoteIndex >= 0 &&
        notes[this.props.activeNoteIndex]) {
      return (<NoteEditor />)
    }

    return (
      <div className='center-text middle-page'>
        <h4>
          {isAtLeastOneNote
            ? t('homeContent:addCreateNote')
            : t('homeContent:thereIsNoNotesYetCreate')}
        </h4>
      </div>
    )
  }
}

HomeContent.propTypes = {
  activeNoteIndex: PropTypes.number.isRequired,
  notes: PropTypes.array.isRequired,
  t: PropTypes.func.isRequired
}
