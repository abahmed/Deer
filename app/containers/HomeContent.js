import { connect } from 'react-redux'
import { compose } from 'redux'
import { withNamespaces } from 'react-i18next'
import HomeContent from '../components/HomeContent'

const mapStateToProps = state => ({
  activeNoteIndex: state.noteReducer.activeNoteIndex
})

const mapDispatchToProps = dispatch => ({
})

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withNamespaces()
)(HomeContent)
