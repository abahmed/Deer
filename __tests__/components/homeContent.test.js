import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import HomeContent from '../../app/components/HomeContent'

jest.mock('../../__mocks__/electron')

const defaultProps = {
  activeNoteIndex: -1,
  t: (key) => key,
  notes: []
}

describe('<HomeContent />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const props = { ...defaultProps, notes: [{}] }
      const wrapper = shallow(<HomeContent {...props} />)

      expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('renders the component with empty notes list', () => {
      const wrapper = shallow(<HomeContent {...defaultProps} />)

      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})
