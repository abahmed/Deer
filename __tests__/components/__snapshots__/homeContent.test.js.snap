// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeContent /> render() renders the component 1`] = `
<div
  className="center-text middle-page"
>
  <h4>
    homeContent:addCreateNote
  </h4>
</div>
`;

exports[`<HomeContent /> render() renders the component with empty notes list 1`] = `
<div
  className="center-text middle-page"
>
  <h4>
    homeContent:thereIsNoNotesYetCreate
  </h4>
</div>
`;
